sdk/src/client/api/block_builder/mod.rs

```diff
@@ -4,21 +4,27 @@
 pub mod input_selection;
 pub mod transaction;
 
+use crypto::keys::bip44::Bip44;
+
 pub use self::transaction::verify_semantic;
 use crate::{
-    client::{ClientInner, Result},
-    types::block::{core::Block, parent::StrongParents, payload::Payload, signature::Ed25519Signature, IssuerId},
+    client::{secret::SecretManage, ClientInner, Result},
+    types::block::{core::Block, parent::StrongParents, payload::Payload, IssuerId},
 };
 
 impl ClientInner {
-    pub async fn finish_basic_block_builder(
+    pub async fn finish_basic_block_builder<S: SecretManage>(
         &self,
         issuer_id: IssuerId,
-        signature: Ed25519Signature,
         issuing_time: Option<u64>,
         strong_parents: Option<StrongParents>,
         payload: Option<Payload>,
-    ) -> Result<Block> {
+        coin_type: u32,
+        secret_manager: &S,
```

Comment:
I would prefer not to accept a secret manager here (unless maybe if we split finish_basic_block_builder(), so one function prepares the final block, just without the signature and then the signing happens in another transaction)
Also wherever we do, we shouldn't just accept the coin_type, but `Bip44`, since a user will probably not always want the public key from the default path (account and address index 0)

Reply:
seems kind of pointless to split it up, but the other point is fair